hiera/test/hiera/settings.yaml

```diff
@@ -1,77 +1,80 @@
-    root_password: 'password'
-    log_path: '/log/database'
-
-## application logging: the possible logging levels are constrained by the
-#                      python 'logging' module.
-#
-#  @LOG_LEVEL, defines the application baseline (i.e. handler) level for logs.
-#      This means the logger level cannot exceed this baseline.
-#
-#      The following are supported 'LOG_LEVEL' values:
-#
-#      - ERROR
-#      - WARNING
-#      - INFO
-#      - DEBUG
-#
-#  Note: the specific log levels can be reviewed:
-#
-#        https://docs.python.org/2/library/logging.html#logging-levels
-#
-#  Note: when referencing a hash path value, remember to prefix the call with
-#       the above 'general: root' definition.
-#
-application:
-    model_type:
-        - svm
-        - svr
-    log_level: 'DEBUG'
-    error_log_path: '/log/application/error'
-    warning_log_path: '/log/application/warning'
-    info_log_path: '/log/application/info'
-    debug_log_path: '/log/application/debug'
+## @settings
+#
+#  This file contains configuration settings for the following:
+#
+#    - redis
+#    - database
+#    - application logging
+#
+
+## variables
+general:
+    environment: 'docker'
+    vagrant_implement: false
+    root: '/var/machine-learning'
+    host: 'localhost'
+    user: 'root'
+    group: 'root'
+
+## redis
+redis:
+    bind_address: '0.0.0.0'
+    host: 'redis'
+    port: 6379
+    path: '/etc/redis/redis.conf'
+
+## webserver
+webserver:
+    flask_log_path: '/log/webserver/flask.log'
+
+## application database
+#
+#  @host, defined from executed 'docker' command.
+#
+#  Note: when referencing a hash path value, remember to prefix the call with
+#       the above 'general: root' definition.
+#
+database:
+    bind_address: '0.0.0.0'
+    host: 'database'
+    allow_host: '%'
+    name: 'db_machine_learning'
+    provisioner: 'provisioner'
+    provisioner_password: 'password'
+    username: 'authenticated'
+    password: 'password'
+    root_password: 'password'
+    log_path: '/log/database'
+
+## application logging: the possible logging levels are constrained by the
+#                      python 'logging' module.
+#
+#  @LOG_LEVEL, defines the application baseline (i.e. handler) level for logs.
+#      This means the logger level cannot exceed this baseline.
+#
+#      The following are supported 'LOG_LEVEL' values:
+#
+#      - ERROR
+#      - WARNING
+#      - INFO
+#      - DEBUG
+#
+#  Note: the specific log levels can be reviewed:
+#
+#        https://docs.python.org/2/library/logging.html#logging-levels
+#
+#  Note: when referencing a hash path value, remember to prefix the call with
+#       the above 'general: root' definition.
+#
+application:
+    model_type:
+        - svm
+        - svr
+    log_level: 'DEBUG'
+    error_log_path: '/log/application/error'
+    warning_log_path: '/log/application/warning'
+    info_log_path: '/log/application/info'
+    debug_log_path: '/log/application/debug'
+
+crypto:
+    salt_length: 32

```

Comment:
You've implemented `CRYPTO_PATH=settings['crypto']['path']` in [`factory.py`](https://github.com/jeff1evesque/machine-learning/pull/2707/files#diff-ab2c5ef35603caa654ec40f8cad9162aR78).  However, you haven't defined it under `salt_length`.  This may be why you're getting `create_app` related [errors](https://travis-ci.org/jeff1evesque/machine-learning/builds/157919970#L3191-L3560) during the travis ci unit testing.